src/View/Form/EntityContext.php

```diff
@@ -372,6 +372,74 @@ public function entity($path = null)
         ));
     }
 
+    /**
+     * Fetch the terminal or leaf entity for the given path.
+     *
+     * Traverse the path until an entity cannot be found. Lists containing
+     * entities will be traversed if the first element contains an entity.
+     * Otherwise the containing Entity will be assumed to be the terminal one.
+     *
+     * @param array|null $path Each one of the parts in a path for a field name
+     *  or null to get the entity passed in constructor context.
+     * @return array Containing the found entity, and remaining un-matched path.
+     * @throws \RuntimeException When properties cannot be read.
+     */
+    protected function leafEntity($path = null)
+    {
+        if ($path === null) {
+            return $this->_context['entity'];
+        }
+
+        $oneElement = count($path) === 1;
+        if ($oneElement && $this->_isCollection) {
+            throw new RuntimeException(sprintf(
+                'Unable to fetch property "%s"',
```

Comment:
should this explain why it was not possible? or this is caught by the framework in all cases?

Reply:
Right now I catch this exception all the time. I used an exception instead of adding a 'dirty' return value of `[false, []]`. Tuple returns values don't have great support in PHP so I didn't want to add more type variance.